src/app/src/components/DrawTools/EditPolygonModal.js

```diff
@@ -14,15 +14,30 @@ import {
 } from '@chakra-ui/react';
 import { CloudUploadIcon } from '@heroicons/react/outline';
 
-export default function EditPolygonModal({ isOpen, defaultLabel, onClose }) {
-    const [label, setLabel] = useState(defaultLabel);
+import { ACCEPT_SHAPES } from '../../constants';
+import { useBoundaryId, useFilePicker } from '../../hooks';
+import { useReplaceBoundaryShapeMutation } from '../../api/boundaries';
 
-    useEffect(() => setLabel(defaultLabel), [defaultLabel]);
+export default function EditPolygonModal({ isOpen, label, onClose }) {
+    const [isLoading, setIsLoading] = useState(false);
+    const [replaceShape] = useReplaceBoundaryShapeMutation();
```

Comment:
Mutations return meta data about the request as the second object in the returned tuple. Among these data is `isLoading`:
```js
 const [replaceShape, {isLoading}] = useReplaceBoundaryShapeMutation();
```

Reply:
[0febfad](https://github.com/azavea/iow-boundary-tool/pull/175/commits/0febfada8f8e04cd2dabbf7902bf9c6ba8eb9943)